init-tools.sh

```diff
@@ -22,47 +22,59 @@ case $OSName in
         ;;
 esac
 
 if [ ! -e $__PROJECT_JSON_FILE ]; then
- if [ -e $__TOOLRUNTIME_DIR ]; then rm -rf -- $__TOOLRUNTIME_DIR; fi
+    if [ -e $__TOOLRUNTIME_DIR ]; then rm -rf -- $__TOOLRUNTIME_DIR; fi
 
- if [ ! -e $__DOTNET_PATH ]; then
-    # curl has HTTPS CA trust-issues less often than wget, so lets try that first.
-    which curl > /dev/null 2> /dev/null
-    if [ $? -ne 0 ]; then
-      mkdir -p "$__DOTNET_PATH"
-      wget -q -O $__DOTNET_PATH/dotnet.tar https://dotnetcli.blob.core.windows.net/dotnet/dev/Binaries/${__DOTNET_TOOLS_VERSION}/${__DOTNET_PKG}.${__DOTNET_TOOLS_VERSION}.tar.gz
-    else
-      curl -sSL --create-dirs -o $__DOTNET_PATH/dotnet.tar https://dotnetcli.blob.core.windows.net/dotnet/dev/Binaries/${__DOTNET_TOOLS_VERSION}/${__DOTNET_PKG}.${__DOTNET_TOOLS_VERSION}.tar.gz
-    fi
-    cd $__DOTNET_PATH
-    tar -xf $__DOTNET_PATH/dotnet.tar
-    if [ -n "$BUILDTOOLS_OVERRIDE_RUNTIME" ]; then
-        find $__DOTNET_PATH -name *.ni.* | xargs rm 2>/dev/null
-        cp -R $BUILDTOOLS_OVERRIDE_RUNTIME/* $__DOTNET_PATH/bin
-        cp -R $BUILDTOOLS_OVERRIDE_RUNTIME/* $__DOTNET_PATH/bin/dnx
-        cp -R $BUILDTOOLS_OVERRIDE_RUNTIME/* $__DOTNET_PATH/runtime/coreclr
-    fi
+    if [ ! -e $__DOTNET_PATH ]; then
+        echo "Installing dotnet cli..."
+        # curl has HTTPS CA trust-issues less often than wget, so lets try that first.
+        which curl > /dev/null 2> /dev/null
+        if [ $? -ne 0 ]; then
+            mkdir -p "$__DOTNET_PATH"
+            wget -q -O $__DOTNET_PATH/dotnet.tar https://dotnetcli.blob.core.windows.net/dotnet/dev/Binaries/${__DOTNET_TOOLS_VERSION}/${__DOTNET_PKG}.${__DOTNET_TOOLS_VERSION}.tar.gz
+        else
+            curl -sSL --create-dirs -o $__DOTNET_PATH/dotnet.tar https://dotnetcli.blob.core.windows.net/dotnet/dev/Binaries/${__DOTNET_TOOLS_VERSION}/${__DOTNET_PKG}.${__DOTNET_TOOLS_VERSION}.tar.gz
+        fi
+        cd $__DOTNET_PATH
+        tar -xf $__DOTNET_PATH/dotnet.tar
+        if [ -n "$BUILDTOOLS_OVERRIDE_RUNTIME" ]; then
+            find $__DOTNET_PATH -name *.ni.* | xargs rm 2>/dev/null
+            cp -R $BUILDTOOLS_OVERRIDE_RUNTIME/* $__DOTNET_PATH/bin
+            cp -R $BUILDTOOLS_OVERRIDE_RUNTIME/* $__DOTNET_PATH/bin/dnx
+            cp -R $BUILDTOOLS_OVERRIDE_RUNTIME/* $__DOTNET_PATH/runtime/coreclr
+        fi
 
-    cd $__scriptpath
- fi
+        cd $__scriptpath
+    fi
 
- mkdir "$__PROJECT_JSON_PATH"
- echo $__PROJECT_JSON_CONTENTS > "$__PROJECT_JSON_FILE"
+    mkdir "$__PROJECT_JSON_PATH"
+    echo $__PROJECT_JSON_CONTENTS > "$__PROJECT_JSON_FILE"
 
- if [ ! -e $__BUILD_TOOLS_PATH ]; then
-    $__DOTNET_CMD restore "$__PROJECT_JSON_FILE" --packages $__PACKAGES_DIR --source $__BUILDTOOLS_SOURCE
- fi
+    if [ ! -e $__BUILD_TOOLS_PATH ]; then
+        echo "Restoring BuildTools version $__BUILD_TOOLS_PACKAGE_VERSION..."
+        $__DOTNET_CMD restore "$__PROJECT_JSON_FILE" --packages $__PACKAGES_DIR --source $__BUILDTOOLS_SOURCE
```

Comment:
That's a good suggestion, but I'd prefer that to be done in a separate PR. If you want to log an issue for it, that would be great.